src/test_hygiene/pyfunction.rs

```diff
@@ -1,12 +1,18 @@
 #![no_implicit_prelude]
-#![allow(unused_variables, clippy::unnecessary_wraps)]
+#![allow(unused_variables, clippy::unnecessary_wraps, dead_code)]
 
 #[crate::pyfunction]
 #[pyo3(crate = "crate")]
 fn do_something(x: i32) -> crate::PyResult<i32> {
     ::std::result::Result::Ok(x)
 }
 
+#[crate::pyfunction]
```

Comment:
I understand that the compile-time test is already valuable, but adding a `#[test]` which calls this seem preferable to allowing dead code IMHO as we also make sure that the result of expanding the macro works at least in a simple case?

Reply:
To be honest, I wonder if there's much value at all in adding this test? I guess it proves that we've fixed the immediate issue, but it's not going to help if we change the macro output I

... I'm not sure if there's anything which can be done here apart from the learning "don't shadow obvious variable names in the macro code", which we reviewers may just have to keep an eye out for manually?